src/freon.c

```diff
@@ -152,24 +154,44 @@ int ioctl(int fd, unsigned long int request, ...) {
     return ret;
 }
 
+static int _open(int (*origfunc)(const char *pathname, int flags, mode_t mode),
+                 const char *origname, const char *pathname, int flags, mode_t mode);
+
 int open(const char *pathname, int flags, ...) {
     if (!orig_open) preload_init();
 
     va_list argp;
```

Comment:
If you pass &flags to _open you can pull the shared vaargs stuff into the shared function as well